db/migrate/20240206150348_create_comparison_metric_models.rb

```diff
@@ -0,0 +1,80 @@
+class CreateComparisonMetricModels < ActiveRecord::Migration[6.1]
+  def change
+    create_table :comparison_periods do |t|
+      t.string :label, null: false # we might want to translate this?
+      t.date :start_date, null: false
+      t.date :end_date, null: false
+
+      t.timestamps
+    end
+
+    create_table :comparison_metrics do |t|
+      t.references :school, null: false, foreign_key: { on_delete: :cascade }
+      t.references :metric_type, null: false, foreign_key: { to_table: 'comparison_metrics', on_delete: :cascade }
+      t.references :alert_type, null: false
+
+      t.string :value # type can vary, how do we do this?????
+
+      t.integer :reporting_period # enum
+      # thought splitting the periods in to two might provide more flexibility. I might be wrong but is
+      # it true that we're not always comparing to a previous period? and that a metric can just be relevant to
+      # just a single period?
+      # is there the option to store exact date information for every metric rather than just for custom ones?
+      # This could be then provide the user with richer information of the exact period the figure is relevant to
+      t.references :custom_period, foreign_key: { to_table: 'comparison_periods', on_delete: :cascade }
+      t.references :custom_previous_period, foreign_key: { to_table: 'comparison_periods', on_delete: :cascade }
+
+      t.boolean :enough_data, null: false
+      t.boolean :whole_period, null: false
+      t.boolean :recent_data, null: false
+      t.date :asof_date
+
+      t.timestamps
+    end
+
+    # This was the table suggested in the design - see above for alternative suggestion
+    # create_table :comparison_periods do |t|
+    #   t.current_period_start_date :date
+    #   t.current_period_end_date :date
+    #   t.previous_period_start_date :date
+    #   t.previous_period_end_date :date
+    #   t.current_period_label :string
+    #   t.previous_period_label :string
+    #   t.timestamps
+    # end
+
+# Jumped a bit ahead here
+
+    create_table :comparison_reports do |t|
+      t.string :key, null: false, unique: true
+      ## translated: title, null: false
+      ## rich text: introduction
+      ## rich text: notes
+
+      t.boolean :public, default: false
+      t.integer :reporting_period # enum
+      # for custom reports
+      t.references :custom_period, null: false, foreign_key: { to_table: 'comparison_periods', on_delete: :cascade }
+      t.references :custom_previous_period, foreign_key: { to_table: 'comparison_periods', on_delete: :cascade }
+      t.timestamps
+    end
+
+=begin
+    # How do we tie metrics to reports?
```

Comment:
I'd assumed this would be in the code, rather than configured. A report may end up with multiple tables, which a range of metrics. So not explicit association in the database.